scalableminds/webknossos
# Add folders to dataset table

This PR adds the child folders of the currently active folder in the dashboard to the dataset table. This pr should include all changes mentioned in [#6883](https://github.com/scalableminds/webknossos/issues/6883), except for the drag & drop feature. I disabled this for the folders as suggested to maybe not include this in the first iteration. Although if you think this should already be included, I can do it. I do not have an opinion on this except that leaving this feature away makes this pr earlier merge ready

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### TODOs: 
- [x] Open the folder settings upon clicking on some icon in the actions column for folders
- [x] Implement context menu support for the folder rows
- [x] disable drag & drop for the folders
- [x] Decrease the width of the types column
- [x] The settings actions for a folder should only be available to an admin or such.

Optionally:
- ~~[ ] allow dragging dataset rows into a folder row~~
  - Not so important IMO: one can still drag the rows to the left sidebar and drop them there into the respective folder.

### Bugs
- [x] The expand icon should not be rendered for the folders
- [x] The details for the folders seem broken

### Steps to test:
- Open the dataset table in the dashboard
- Select the root folder in the left folder tree view
- Right-click on the root folder and create a new folder. The new folder should be listed in the table in the center.
- creating another folder should also result in the folder appearing in the table.
- Right-clicking on a folder in the table should open up the same context menu as if one would click on the folder in the folder hierarchy view on the left.
- single clicking a folder should show its details on the right (another single click should deselect it)
- selecting multiple folders should not work
- drag & drop of folders in the table should not work
- double-clicking a folder should open that folder's contents in the table
- If logged in as an unprivileged user (non admin / dataset manager) manipulating any folder setting should not work / not be available in the UI or the UI should be disabled.
- Please look at [#6883](https://github.com/scalableminds/webknossos/issues/6883) for more details on how the folder rows should behave. (But I think I covered everything in the list above)

### Issues:
- fixes #6883

------
(Please delete unneeded items, merge only when none are left open)
